src/index.tsx

```diff
@@ -0,0 +1,13 @@
+import ReactDOM from "react-dom/client";
+import App from "./App";
+
+// const root = ReactDOM.createRoot(document.getElementById("root"));
+// root.render(<App />);
+
+const rootElement = document.getElementById("root");
+if (rootElement) {
+  const root = ReactDOM.createRoot(rootElement);
+  root.render(<App />);
+} else {
+  console.error("Root element not found in the document.");
+}
```

Comment:
```suggestion
import ReactDOM from "react-dom/client";
import App from "./App";

const root = ReactDOM.createRoot(document.getElementById("root") as HTMLElement);
root.render(<App />);
```

이렇게 바꾸시면 타입 에러 해결될거에요!